snuba/datasets/plans/query_plan.py

```diff
@@ -34,26 +40,36 @@ class StorageQueryPlan:
     # having two query objects of the same type around during processing
     # would be dangerous, so it is probably better not to expose the query
     # here yet.
-    query_processors: Sequence[QueryProcessor]
+    plan_processors: Sequence[QueryProcessor]
+    db_query_processors: Sequence[QueryProcessor]
     execution_strategy: QueryPlanExecutionStrategy
```

Comment:
From what I can tell, the `db_query_processors` (referring to them below as "post-split" processors even though that's not completely accurate) are only used as an attribute here as a means of threading them through to `QueryPlanExecutionStrategy.execute`.

Considering that:

1. the execution strategy is already exclusively responsible for running the post-split processors,
2. no other component can mutate the list of processors (at least as implied by this being a non-mutable `Sequence`), and
3. the execution strategy instance itself is constructed alongside the `StorageQueryPlan` in the `StorageQueryPlanBuilder`, implying that the builder has intimate knowledge about how the query will be ran (which is kind of obvious given the name)

It seems to me that the post-split processors are tightly coupled to the `QueryPlanExecutionStrategy` — should they just be instance attributes on that class, obviating the need for the `QueryPlanExecutionStrategy.execute` changes? Or is there some other benefit to this approach?

Reply:
Indeed, there is no functional need for the post-split processors to be passed to the execution strategy through the execute method. I had considered the option of making them an attribute of the execution strategy itself instead. 
The only reason I moved in this direction was to make it explicit in the contract the execution strategy provides that it needs to take them in and needs to use them and that has to be true for any implementation of the execution strategy abstract class.
If we make them attributes only, it is up to the implementations (the abstract class would not have any trace of them). This constraint is more implicit.
Said that, the risk of misuse is quite low since we would have to mess up both the query plan builder and the execution strategy, so I am open to change this to make them attributes if you feel the improvement in the interface clarity outweighs the risk.

Reply:
Okay, that makes sense. I guess either approach has its positives and negatives — I think that making them an instance attribute on `QueryPlanExecutionStrategy` makes the control flow and responsibilities easier to understand at a glance since it doesn't require as much knowledge of the overall execution staging (it also seems sort of appropriate, since the variation in the way the execution strategies work motivated this change in the first place; I also think the variable naming would seem more natural) but it does seem like it gives us more leeway to shoot ourselves in the foot than what we'd ideally have. I'll leave the decision about which approach makes the most sense up to you. If you decide to review it please dismiss this review and re-request, though.

Reply:
Moved the processors into the Execution Strategy as attribute. 
The docstring of the QueryPlanExecutionStrategy seems to me clear enough on the responsibilities of the implementations, so I am ok this way.